source/MRMesh/MREdgePaths.cpp

```diff
@@ -547,46 +550,60 @@ void dilateRegionByMetric( const MeshTopology & topology, const EdgeMetric & met
 
     while ( !builder.done() && builder.doneDistance() <= dilation )
     {
+        if ( callback && !callback( builder.doneDistance() / dilation ) )
```

Comment:
It is typically too expensive to invoke callback in every iteration, please do it e.g. only once per 1024 iterations